doc/plugin_agent_nodeattestor_k8s_psat.md

```diff
@@ -7,10 +7,11 @@ reads and provides the signed projected service account token (PSAT) to the serv
 In addition to service account data, PSAT embeds the pod name and UID on its claims. This allows
 SPIRE to create more fine-grained attestation policies for agents.
 
-The plugin generates SPIFFE IDs with the form:
+The plugin queries Kubernetes API server to obtain the node in which the agent is running on.
+In this way, it generates SPIFFE IDs with the form:
 
 ```
-spiffe://<trust domain>/spire/agent/k8s_psat/<cluster>/<POD UID>
+spiffe://<trust domain>/spire/agent/k8s_psat/<cluster>/<node name>
```

Comment:
Node name is unique at a given time. It could be deleted and created again holding the same name and a new UID. However, two nodes with the same name and different UID can't exist at the same time.

From k8s docs:
> Only one object of a given kind can have a given name at a time. However, if you delete the object, you can make a new object with the same name.

https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names

Reply:
Hmmm I didn't know that node has a UID. Thoughts on using node UID instead of name? It would technically be a different node at that point, right? Can we get node UID from kubelet? In this case, node name seems more appropriate as a selector than as the agent ID

Sorry for the back and forth on this :-/

Reply:
No worries! Some thoughts:
- If we use node UID, the spiffeID would be unique, and we will get a new spiffeID every time the node is recreated.
- If we use node name, the agent will re-attest every time the node is recreated, and the spiffe ID will be kept.

After some research, it looks like there is not possible to get node UID from the kubelet. I haven't been able to find documentation about kubelet API but looking into [code](https://github.com/kubernetes/kubernetes/blob/dc071b8e81b33e8f4456ab2edcb03a45a75674ee/pkg/kubelet/server/server.go#L279-L337) it seems that it can not be got.

If we decide to have node UID into spiffeID, I guess we would have to get it from Kubernetes API server

Reply:
Ok. In the interest of moving things along here, I'd vote to just keep the node name rather than take on api server dependency again. Thanks very much for looking into this